base_layer/wallet/src/connectivity_service/service.rs

```diff
@@ -271,48 +281,35 @@ impl WalletConnectivityService {
     }
 
     fn get_base_node_peer_manager(&self) -> Option<BaseNodePeerManager> {
-        self.base_node_watch_receiver.borrow().as_ref().map(|p| p.clone())
+        self.base_node_watch_receiver.borrow().as_ref().cloned()
     }
 
     async fn disconnect_base_node(&mut self, peer_id: PeerId) {
+        if let Some(pool) = self.current_pool.take() {
+            pool.close().await;
+        }
         if let Err(e) = self.network_handle.disconnect_peer(peer_id).await {
             error!(target: LOG_TARGET, "Failed to disconnect base node: {}", e);
         }
-        self.pools.remove(&peer_id);
     }
 
-    async fn setup_base_node_connection(&mut self) {
-        let Some(mut peer_manager) = self.get_base_node_peer_manager() else {
-            debug!(target: LOG_TARGET, "No base node peer manager set");
-            return;
-        };
+    async fn setup_base_node_connection(&mut self, mut peer_manager: BaseNodePeerManager) {
```

Comment:
Every time this function is called, it should attempt to use the next peer on the list as we have an issue with the peer connection at the current index, except if it is the very first time, in which case it should attempt connecting to the peer at the current peer index.

Reply:
I believe this is what is happening - we move to the next peer after attempting the current

Reply:
pushed